crates/noirc_evaluator/src/ssa/block.rs

```diff
@@ -94,6 +97,35 @@ impl BasicBlock {
         }
         result
     }
+
+    //Returns true if the block is a short-circuit
+    fn is_short_circuit(&self, ctx: &SsaContext, assumption: Option<NodeId>) -> bool {
+        if let Some(ass_value) = assumption {
+            if self.instructions.len() >= 3 {
```

Comment:
I agree this seems arbitrary but it is not a problem. If there is a false constrain with assumption matching the condition (which is what the function is checking), then it is a short-circuit, so even if a user can create blocks like this, it is fine to short-circuit the other blocks.

What we may be missing is if there is a short-circuit block which does not start with these instructions but would contain them anyways. This is unlikely to happen because the short_circuit() creates them like this and we do the merge_blocks right after, but even if this happen, the resulting circuit will be correct, we are just missing some possible optimisation opportunity.